src/Pos/Generator/Block/Payload.hs

```diff
@@ -196,7 +197,7 @@ genTxPayload = do
                     splitCoins outputsN (unsafeIntegerToCoin inputsSum)
                 let txOuts = NE.fromList $ zipWith TxOut outputAddrs coins
                 let txOutAuxsPre = map (\o -> TxOutAux o []) txOuts
-                either (lift . throwM . BGFailedToCreate) pure =<<
+                either (lift . throwM . BGFailedToCreate . unTxError) pure =<<
```

Comment:
I think it's more correct to attach `TxError` to `BGFailedToCreate`. If later we extend `TxError`, we will want this information in block generator too.